src/composed/message/parser.rs

```diff
@@ -160,8 +160,10 @@ pub(super) fn next(
                     bail!("missing next packet");
                 };
 
-                let reader =
-                    SignatureOnePassReader::new(&one_pass_signature, Box::new(inner_message))?;
+                let reader = SignatureOnePassReader::new(
+                    one_pass_signature.clone(),
+                    Box::new(inner_message),
+                )?;
                 let message = Message::SignedOnePass {
                     one_pass_signature,
```

Comment:
we can avoid the clone, and maybe not pass it in here? but rather read it from the `reader`?